packages/inference/src/lib/makeRequestOptions.ts

```diff
@@ -61,6 +57,27 @@ export async function makeRequestOptions(
 	if (!model) {
 		throw new Error("No model provided, and no default model found for this task");
 	}
+	if (provider) {
+		if (!INFERENCE_PROVIDERS.includes(provider)) {
+			throw new Error("Unknown Inference provider");
+		}
+		if (!accessToken) {
+			throw new Error("Specifying an Inference provider requires an accessToken");
+		}
+		switch (provider) {
+			case "replicate":
+				model = REPLICATE_MODEL_IDS[model];
+				break;
+			case "sambanova":
+				model = SAMBANOVA_MODEL_IDS[model];
+				break;
+			case "together":
+				model = TOGETHER_MODEL_IDS[model]?.id ?? model;
+				break;
+			default:
+				break;
```

Comment:
We might want to throw an erro here if the model is not in the hardcoded `MODEL_IDS` mappings